7_deploy_osh/roles/suse-build-images/tasks/setup-build-registry.yml

```diff
@@ -0,0 +1,63 @@
+---
+
+- name: Ensure docker is running on host
+  service:
+    name: docker
+    state: started
+
+- name: Image Building - Check if local docker registry server is running
+  shell: |
+    docker ps | grep {{ local_registry_name }}
+  register: registry_running_cmd_status
+  failed_when: false
+  tags:
+    - run
+
+- name: Image Building - Clean local docker registry server if already running
+  shell: |
+    docker stop {{ local_registry_name }}
+    docker rm {{ local_registry_name }}
+  when:
+    - registry_running_cmd_status.rc == 0
+
+- name: Image Building - Start insecure local docker registry
+  shell: |
+    docker run -d -p {{ local_registry_port }}:{{ local_registry_port }} --name {{ local_registry_name }} --restart always registry:2
+
+- name: Image Building - Ensure local-registry DNS resolution is set in /etc/hosts
+  blockinfile:
+    path: /etc/hosts
+    block: |
+      {{ ansible_default_ipv4.address | ipaddr('address') }} {{ local_registry_name }}
+
+- name: Image Building - Make sure docker daemon file is present on host
+  copy:
+    dest: "/etc/docker/daemon.json"
+    content: "{}"
+    force: no
+    mode: 0644
+
+- name: Image Building - Read existing docker daemon conf file
+  set_fact:
+    _docker_daemon_conf: "{{ lookup('file', '/etc/docker/daemon.json') | default({}) | from_json }}"
```

Comment:
Isn't this wrong? The `/etc/docker/daemon.json` file is located on remote node, but lookup is looking at local machine, am I right?

Reply:
You are right, that needs changing.

Reply:
We are changing so local docker cli can communicate with insecure registry. For consumption by docker clients in caasp nodes, we are using secure registry. The insecure to secure registry dance is because of bug in loci code which does not work with secure registry when it tries to pull earlier built requirements file. Possibly in following line and I should have fixed it (looks like got distracted in some other task)

https://github.com/openstack/loci/blob/master/scripts/fetch_wheels.py#L61

Reply:
Ok I guess the next steps are:
1) Have a role for secure docker registry configuration
2) Have a role for building images
3) Fix LOCI with secure registry.

I was working on the point 2 last week (as it's in our tasks in the board), and will have a PR soon, when the step 1 is ready.

Reply:
Please ignore this, as the points 1 and 2 I mentionned above have be implemented here: https://github.com/SUSE-Cloud/socok8s/pull/93